torch/csrc/Module.cpp

```diff
@@ -1721,6 +1722,33 @@ Call this whenever a new thread is created in order to propagate values from
         return torch::should_allow_numbers_as_tensors(name);
       });
 
+  // FIXME(crcrpar): Better to have `at::ScalarType` get mapped to `torch.dtype`
+  // Currently I see the second item of the key is displayed as
+  // e.g. `torch._C._te.ScalarType at 0x7fcf318adab0`
+  // I thought adding an appropriate type_caster of `at::ScalarType` to
+  // torch/csrc/pybind.h` would solve this but it caused segmentation fault in
+  // my environment.
+  using _DeviceDtypeKey = std::pair<at::Device, std::string>;
+  using _FlatMap = std::unordered_map<
+      _DeviceDtypeKey,
+      at::native::TensorsAndIndicesT,
+      at::native::ParamsHash<_DeviceDtypeKey>>;
+  py_module.def(
+      "_group_tensors_by_device_and_dtype",
+      [](const std::vector<std::vector<c10::optional<at::Tensor>>>&
+             nested_tensorlist,
+         const bool with_indices) {
+        _FlatMap map;
+        for (const auto& iter :
+             at::native::_group_tensors_by_first_tensors_device_and_dtype(
+                 nested_tensorlist, with_indices)) {
+          const auto scalar_type_name =
+              torch::utils::getDtypeNames(iter.first.second).first;
+          map.insert({{iter.first.first, scalar_type_name}, iter.second});
```

Comment:
The internal error seems to point to here--@crcrpar do you have an inkling of why error C2227 (left of '->{ctor}' must point to class/struct/union/generic type) would occur here? the thing to the left seems to be `iter`

Reply:
hmm or is it map...

Reply:
Oooooh it is the constructor for the {}, I see. I'm not a C++ expert--I wonder if being explicit about the constructor will make this error go away

Reply:
I have Win machine I can quickly test the changes on. Will try to kick the build this afternoon

Reply:
thank you @malfet

Reply:
I feel like it's a particular build of windows though...the CI window builds on ciflow/trunk passed

Reply:
Yes, it's an internal VS2019 debug build (for posterity, one can trigger it using `buck build @arvr/mode/win/vs2019/dev //xplat/caffe2:_C_impl`

Reply:
Unless I am misunderstanding, we are trying to insert using an initializer list where the elements are `std::pair` of `_DeviceDtypeKey` and `at::native::TensorsAndIndicesT`... So `{{iter.first.first, scalar_type_name}, iter.second}` would be an element of the initializer list? So it would seem there should be another set of braces, i.e. `{{{iter.first.first, scalar_type_name}, iter.second}}`.

It's signature 4 on the [docs for unordered_map:insert on cplusplus.com](https://cplusplus.com/reference/unordered_map/unordered_map/insert/).

Might be more clear using one of the other signatures😵‍💫